utils/build-script-impl

```diff
@@ -1081,40 +1083,20 @@ if [[ ! "${SKIP_BUILD_PLAYGROUNDSUPPORT}" && ! -d ${PLAYGROUNDSUPPORT_SOURCE_DIR
     exit 1
 fi
 
-PRODUCTS=(cmark llvm)
-if [[ ! "${SKIP_BUILD_LIBCXX}" ]] ; then
-     PRODUCTS=("${PRODUCTS[@]}" libcxx)
-fi
-if [[ ! "${SKIP_BUILD_LIBICU}" ]] ; then
-     PRODUCTS=("${PRODUCTS[@]}" libicu)
-fi
-PRODUCTS=("${PRODUCTS[@]}" swift)
-if [[ ! "${SKIP_BUILD_LLDB}" ]] ; then
-     PRODUCTS=("${PRODUCTS[@]}" lldb)
-fi
-# LLBuild and XCTest are dependent on Foundation, so Foundation must
+PRODUCTS=( cmark llvm )
+[[ "${SKIP_BUILD_LIBCXX}" ]] || PRODUCTS+=( libcxx )
```

Comment:
Shouldn't these be inverted?

Reply:
I believe that this is correct: you either skip the build OR you build the product.  I suppose we could write this as `[[ ! "${SKIP_BUILD_LIBCXX}" ]] && PRODUCTS+=( libcxx )`.  They should be logically equivalent.

Reply:
Oh, silly me. I spaced on the operator used here. Either is fine.